### PR TITLE
Fixed bug where calling AppendBlockFromUri without Options bag throws a null exception error; Also cleaning up openwrite records

### LINKED ISSUE #29062: [BUG] AppendBlobClient.AppendBlockFromUri() with no options argument throws NullReferenceException
### Library name and version

Azure.Storage.Blobs 12.12.0

### Describe the bug

Omitting the `AppendBlobAppendBlockFromUriOptions options` argument or passing it as `null` in  

```
AppendBlobClient.AppendBlockFromUri(
        Uri sourceUri, 
        AppendBlobAppendBlockFromUriOptions options = default, 
        CancellationToken cancellationToken = default)
```
throws a `System.NullReferenceException`.

### Expected behavior

Omitting or passing the `options` argument as `null` should pick up default values (as options properties are optional)

### Actual behavior

The `AppendBlobClient.AppendBlockFromUri()` method call throws a `System.NullReferenceException` when `options` argument is `null`

### Reproduction Steps

```
appendBlobClient.AppendBlockFromUri(uri);
```

### Environment

```
.NET SDK
        Version:   6.0.300
        Commit:    8473146e7d

Runtime environment:
        OS Name:     Mac OS X
        OS Version:  12.2
        OS Platform: Darwin
        RID:         osx.12-x64
        Base Path:   /usr/local/share/dotnet/sdk/6.0.300/

Host (useful for support):
        Version: 6.0.5
        Commit:  70ae3df4a6
```

### PR DESCRIPTION
**Pull request by amnguye**
Looking to resolve https://github.com/Azure/azure-sdk-for-net/issues/29062

### CODE REVIEW

**Comment by amnguye** [comment]
Also looks like a script ran and cleaned up unused TestRecords. Most of these test records have moved to our OpenWrite folder

File: sdk/storage/Azure.Storage.Blobs/src/AppendBlobClient.cs [hunk 1]
```diff
@@ -1386,11 +1386,11 @@ public new AppendBlobClient WithEncryptionScope(string encryptionScope)
             CancellationToken cancellationToken = default) =>
             AppendBlockFromUriInternal(
                 sourceUri,
-                options.SourceRange,
-                options.SourceContentHash,
-                options.DestinationConditions,
-                options.SourceConditions,
-                options.SourceAuthentication,
+                options != default ? options.SourceRange : default,
```
Comment (jaschrep-msft) [inline sdk/storage/Azure.Storage.Blobs/src/AppendBlobClient.cs:1389]:
```suggestion
                options?.SourceRange ?? default
```

**Review by jaschrep-msft — APPROVED** [review]
nit readability fix, otherwise good

**Review by seanmcc-msft — APPROVED** [review]

**Comment by seanmcc-msft** [comment]
Looks good!

**Pull request merged**